app.rb

```diff
@@ -50,4 +50,43 @@ def debug_params
     list.save!
   end
   redirect back
-end
\ No newline at end of file
+end
+
+#----------------- Add New List ------------------
+
+post "/newlist/name" do
+  new_list = List.new("#{params[:id]}")
+  new_list.name = "#{params[:id]}" 
+  new_list.save!
+  List.new("#{new_list.name}")
+  #new_list.filename
+  redirect back
+end
+
+
+#----------------- Delete a List ------------------
+
+post "/delist/name" do
+  list_dump = List.new("#{params[:id]}")
+  target = list_dump.id.concat(".md")
+  files = Dir["data/*.md"]
+  files.each do |file|
+    file.delete("target")
```

Comment:
file is just a string here so `file.delete("target")` here does not have any effect. let's talk about list deletion in person next time and i'd be happy to walk you through it.

basically you can:
1. delete the file for the list you wanna delete (implement a `List#delete` method
2. call `lists = List.load_all`, then `lists.each_with_index` and set `list.id = index` for each list and save the list